### PR TITLE
Document casting timestamp_tz column to a date as part of Iceberg optimize #12943

### PR DESCRIPTION
**Pull request by Ankan0011**
## Description
Documentation for support Iceberg OPTIMIZE with WHERE casting timestamp_tz column to a date.

## Additional context and related issues
Fixes https://github.com/trinodb/trino/issues/12943

## Release notes

(X) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 1]
```diff
@@ -575,6 +574,19 @@ apply ``optimize`` only on the partitions corresponding to the filter:
     ALTER TABLE test_partitioned_table EXECUTE optimize
     WHERE partition_key = 1
 
+``optimize`` command now supports predicate over base columns. One of the optimization
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/iceberg.rst:577]: Remove "now".

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -575,6 +574,19 @@ apply ``optimize`` only on the partitions corresponding to the filter:
...
+which you can also use with the ``WHERE`` clause is to compare date.
+
+.. code-block:: sql
+
+  ALTER TABLE test_table EXECUTE optimize
+  WHERE CAST(timestamp_tz AS date) > date_constant
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/iceberg.rst:583]: We could replace `date_constant` with the example, e.g. `DATE '2016-08-09'`

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 3]
```diff
@@ -575,6 +574,19 @@ apply ``optimize`` only on the partitions corresponding to the filter:
...
+
+Casting the timestamp(UTC) to DATE first will create the point in DATE datatype.
+This will first rewrites the column according to the DATE casted datatype, then
+the query can be speedup due to the compaction performed on the date column when
+comparison with a constant date.
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/iceberg.rst:588]: Please avoid mentioning the implementation's details. It's unrelated to users.

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 4]
```diff
@@ -1417,4 +1429,4 @@ Table redirection
 .. include:: table-redirection.fragment
 
 The connector supports redirection from Iceberg tables to Hive tables with the
-``iceberg.hive-catalog-name`` catalog configuration property.
\ No newline at end of file
+``iceberg.hive-catalog-name`` catalog configuration property.
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/iceberg.rst:1432]: Please separate a commit for a new line change.

**Comment by ebyhr** [comment]
Please fix the commit title. https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#format-git-commit-messages

File: docs/src/main/sphinx/connector/iceberg.md [hunk 5]
```diff
@@ -593,6 +593,18 @@ ALTER TABLE test_partitioned_table EXECUTE optimize
...
+optimization which you can also use with the ``WHERE`` clause is to compare date.
+
+```sql
+ALTER TABLE test_table EXECUTE optimize
+WHERE CAST(timestamp_tz AS DATE) > DATE '2016-08-09'
+```
+
+Casting the timestamp to DATE and then using the ``optimise`` rewrites the 
+column according to the casted datatype, this speedup the query due to the 
+compaction performed on the date column when comparison with a constant date.
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/iceberg.md:606]: We don't need to mention the detailed flow of the procedure.

File: docs/src/main/sphinx/connector/iceberg.md [hunk 6]
```diff
@@ -593,6 +593,15 @@ ALTER TABLE test_partitioned_table EXECUTE optimize
 WHERE partition_key = 1
 ```
 
+``optimize`` can also be used with ``WHERE`` clause with partitioned base
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:596]:
```suggestion
You can use a more complex ``WHERE`` to narrow down the scope of the ``optimize`` procedure. The following example casts the data to a date and uses a comparison to only optimize partitions with data from the year 2022 or newer:
```

File: docs/src/main/sphinx/connector/iceberg.md [hunk 7]
```diff
@@ -593,6 +593,15 @@ ALTER TABLE test_partitioned_table EXECUTE optimize
 WHERE partition_key = 1
 ```
 
+``optimize`` can also be used with ``WHERE`` clause with partitioned base 
+column with casting. You can benefit from partitions for constant date 
+comparisons.
+
+```sql
+ALTER TABLE test_table EXECUTE optimize
+WHERE CAST(timestamp_tz AS DATE) > DATE '2016-08-09'
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:602]:
```suggestion
WHERE CAST(timestamp_tz AS DATE) > DATE '2021-12-31'
```

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/iceberg.md (hunk 6 above)
Comment (ebyhr) [reply docs/src/main/sphinx/connector/iceberg.md]: > casts the data to a date

@mosabua Is it a typo of "casts the timestamp to a date"?

**Comment by ebyhr** [comment]
> Squased commits.

@Ankan0011 Please fix the commit title.

**Comment by ebyhr** [comment]
> Docs for https://github.com/trinodb/trino/issues/12943

Please fix the PR description as "Fixes #12943" so that we can close the issue automatically when merging this PR.

**Comment by mosabua** [comment]
@ebyhr I updated the text, source formatting, and commit message now. Imho this is ready to go now.

**Comment by mosabua** [comment]
@Ankan0011 please have a look

File: docs/src/main/sphinx/connector/iceberg.md [hunk 8]
```diff
@@ -593,6 +593,16 @@ ALTER TABLE test_partitioned_table EXECUTE optimize
 WHERE partition_key = 1
 ```
 
+You can use a more complex ``WHERE`` clause to narrow down the scope of the
+``optimize`` procedure. The following example casts the timestamp values to
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/iceberg.md:597]: This is a markdown file. No need to repeat backtick.

File: docs/src/main/sphinx/connector/iceberg.md [hunk 9]
```diff
@@ -592,6 +592,14 @@ apply `optimize` only on the partitions corresponding to the filter:
 ALTER TABLE test_partitioned_table EXECUTE optimize
 WHERE partition_key = 1
 ```
+You can use a more complex `WHERE` to narrow down the scope of the 
+`optimize` procedure. The following example casts the data to a date and uses
+a comparison to only optimize partitions with data from the year 2022 or newer.
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/iceberg.md:597]: You changed the sentence to the old state. Please revert and fix only backtick number.

File: docs/src/main/sphinx/connector/iceberg.md [hunk 10]
```diff
@@ -592,6 +592,15 @@ apply `optimize` only on the partitions corresponding to the filter:
 ALTER TABLE test_partitioned_table EXECUTE optimize
 WHERE partition_key = 1
 ```
+You can use a more complex `WHERE` clause to narrow down the scope of the
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:595]: We now have a shared included for optimize .. I think you probably should move this example into that fragment after rebasing master

**Review by mosabua — CHANGES_REQUESTED** [review]

**Comment by mosabua** [comment]
@Ankan0011 can you move the text to the included fragment file as requested please?

**Comment by Ankan0011** [comment]
> @Ankan0011 can you move the text to the included fragment file as requested please?

@mosabua Can you check if I have placed the changes in the right fragment ?

**Review by mosabua — APPROVED** [review]
Looks good now.

**Pull request merged**